Python/compile.c

```diff
@@ -1982,6 +1982,12 @@ compiler_function(struct compiler *c, stmt_ty s, int is_async)
     if (!compiler_decorators(c, decos))
         return 0;
 
+    // fixup decorators to be the first line number internally.
+    if (asdl_seq_LEN(decos) > 0) {
+        expr_ty first_decorator = asdl_seq_GET(decos, 0);
+        c->u->u_firstlineno = first_decorator->lineno;
```

Comment:
Why this is needed? It would be a bit awkward if we externally show everywhere line of `def` as a first line while internally always use the decorator line.

Reply:
To clarify, I think it _may_ be OK if you give a good comment with an explanation.

Reply:
We need to maintain the first line number of the code object to be the line number of the first decorator so that `inspect.getsource` will work correctly. The other alternative would be to change this, so that code objects point to both the line number of the def and the first source line number, but I doubt that would be useful, or worth any additional overhead that would cause. I will add a more detailed comment.